File: types/goal.ts
export type GoalStatus = "Active" | "Completed" | "Archived";

export interface GoalResponse {
  id: number;
  title: string;
  description: string;
  category: string;
  targetDate: string;
  progress: number;
  status: GoalStatus;
  archived: boolean;
}

export type CreateGoalPayload = Omit<GoalResponse, "id" | "status" | "archived" | "progress"> & {
  progress?: number;
};

export type UpdateGoalPayload = Partial<CreateGoalPayload> & { status?: GoalStatus; archived?: boolean };

File: app/schemas/goal.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class GoalCreate(BaseModel):
    title: str
    description: str = ""
    category: str = "General"
    target_date: datetime | None = None


class GoalUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    category: Optional[str] = None
    target_date: Optional[datetime] = None


class GoalResponse(GoalCreate):
    id: int
    progress: int
    status: str
    is_archived: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
